packages/request-client.js/src/api/payment-network/btc/default-providers/blockcypher-com.ts

```diff
@@ -57,28 +57,28 @@ export default class BlockcypherCom implements Types.IBitcoinDetectionProvider {
    * @param eventName Indicates if it is an address for payment or refund
    * @returns Balance with events
    */
-  public parse(addressInfo: any, eventName: Types.EVENTS_NAMES): Types.IBalanceWithEvents {
+  public parse(addressInfo: any, eventName: Types.EVENTS_NAMES): Types.BTCBalanceWithEvents {
     const balance = new bigNumber(addressInfo.total_received).toString();
 
     // Retrieves all the transaction hash of the transactions having as input the current address
     const inputTxHashes = addressInfo.txrefs
       .filter((tx: any) => tx.tx_output_n === -1)
       .map((tx: any) => tx.tx_hash);
 
-    const events: Types.IPaymentNetworkEvent[] = addressInfo.txrefs
+    const events: Types.BTCPaymentNetworkEvent[] = addressInfo.txrefs
       // keep only the transaction with this address as output
       .filter((tx: any) => tx.tx_input_n === -1)
       // exclude the transactions coming from the same address
       .filter((tx: any) => !inputTxHashes.includes(tx.tx_hash))
       .map(
-        (tx: any): Types.IPaymentNetworkEvent => ({
+        (tx: any): Types.BTCPaymentNetworkEvent => ({
+          amount: tx.value.toString(),
```

Comment:
imo, `amount` should stay in `event.parameters` because all the events won't have necessary an amount. (e.g "deny a payment")

Reply:
This may need to be discussed with the team

Reply:
discussed with Vince, we agreed that it's OK to have an amount of 0 for specific events.